code/modules/paperwork/ticketmachine.dm

```diff
@@ -48,17 +48,18 @@
 	update_icon()
 
 /obj/machinery/ticket_machine/proc/increment()
-	if(current_number >= ticket_number)
+	if(current_number > ticket_number)
 		return
-	playsound(src, 'sound/misc/announce_dig.ogg', 50, 0)
 	if(current_number && !(obj_flags & EMAGGED) && tickets[current_number])
-		tickets[current_number].visible_message("<span class='notice'>\the [tickets[current_number]] disperses!</span>")
+		to_chat(viewers(3,get_turf(tickets[current_number].loc)), "<span class='notice'>\the [tickets[current_number]] disperses!</span>")
 		qdel(tickets[current_number])
-	current_number ++ //Increment the one we're serving.
-	say("Now serving ticket #[current_number]!")
-	if(!(obj_flags & EMAGGED) && tickets[current_number])
-		tickets[current_number].visible_message("<span class='notice'>\the [tickets[current_number]] vibrates!</span>")
-	update_icon() //Update our icon here rather than when they take a ticket to show the current ticket number being served
+	if(current_number < ticket_number)
+		current_number ++ //Increment the one we're serving.
+		playsound(src, 'sound/misc/announce_dig.ogg', 50, 0)
+		say("Now serving ticket #[current_number]!")
+		if(!(obj_flags & EMAGGED) && tickets[current_number])
+			to_chat(viewers(3,get_turf(tickets[current_number].loc)), "<span class='notice'>\the [tickets[current_number]] vibrates!</span>")
```

Comment:
why not use visible_message on the ticket

Reply:
er sorry I mean audible message

Reply:
@optimumtact works now, thanks